client/verta/tests/test_concurrency/test_concurrency.py

```diff
@@ -0,0 +1,33 @@
+import six
+import multiprocessing
+import pytest
+from .run_functions import log_observation, upload_artifact
+
+from operator import itemgetter
+from functools import partial
+
+
+class TestConcurrency:
+    @pytest.mark.skipif(six.PY2, reason="multiprocessing.Pool has issues in Python 2")
+    def test_multiple_runs_log_obs(self, client):
+        client.set_project()
+        client.set_experiment()
+
+        with multiprocessing.Pool(36) as pool:
+            result = pool.map(partial(log_observation, client), range(10))
+
+            def extract_obs_value(obs):
+                return list(map(itemgetter(0), obs))
+
+            assert all(map(lambda res: extract_obs_value((res["run"].get_observation("obs"))) == res["obs"], result))
```

Comment:
Could you do me a favor and let me know what `pytest`'s stack trace looks like if this fails (maybe hard-code a bad value returned by the parallel function)? My suspicion is that the error message would be absolutely massive because pytest afaik tries to print the value of every single variable in the `assert` statement.

If it _is_ excessively verbose, I'd recommend first assigning the `list(map(...))` to a separate variable,
then `assert all(var)`

Reply:
You're right. Fixed.